File: go-api/internal/models/account.go
package models

// Account represents a user account in the system.
type Account struct {
	// @Description Unique identifier of the account (UUID).
	// @Format uuid
	// @Example 550e8400-e29b-41d4-a716-446655440000
	ID string `json:"id" db:"id"`

	// @Description Unique username chosen by the account owner. Must be unique across the system.
	// @MinLength 3
	// @MaxLength 32
	// @Example charlie
	Username string `json:"username" db:"username"`

	// @Description Timestamp when the account was created (UTC, RFC3339 format).
	// @Format date-time
	// @Example 2025-09-22T19:15:24.526505Z
	CreatedAt string `json:"created_at" db:"created_at"`

	// @Description Timestamp of the last account update (UTC, RFC3339 format).
	// @Format date-time
	// @Example 2025-09-22T19:15:24.526505Z
	UpdatedAt string `json:"updated_at" db:"updated_at"`
}

File: rust-processor/src/models/account.rs
use chrono::{DateTime, Utc};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, FromRow)]
pub struct DbAccount {
    pub id: Uuid,
    pub username: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, FromRow)]
pub struct Account {
    pub id: Uuid,
    pub balance_cents: i64,
}
